service/history/timerQueueProcessorBase.go

```diff
@@ -590,40 +585,45 @@ func (t *timerQueueProcessorBase) processDeleteHistoryEvent(task *persistence.Ti
 		return nil
 	}
 
-	err = t.deleteWorkflowExecution(task)
+	clusterArchivalStatus := t.shard.GetService().GetClusterMetadata().ArchivalConfig().GetArchivalStatus()
+	domainCacheEntry, err := t.historyService.getActiveDomainEntry(common.StringPtr(task.DomainID))
 	if err != nil {
 		return err
 	}
+	domainArchivalStatus := domainCacheEntry.GetConfig().ArchivalStatus
 
-	err = t.deleteWorkflowHistory(task, msBuilder)
-	if err != nil {
-		return err
+	switch clusterArchivalStatus {
```

Comment:
can you emit metrics to get an idea of how many times we (1) delete_history here (2) send signal to archival and delete history later (3) do nothing ?